### DISCUSSION TITLE
进行轨迹数据清洗时报错

### DISCUSSION THREAD

**Discussion by Murphy666-code** [zdsjjtTLG/TrackIt#16] [Q&A]
D:\Anaconda\envs\jiaotong\python.exe "C:\Users\Sha mofei\PycharmProjects\pythonProject\数据读取\gpt坐标系转换.py" 
Traceback (most recent call last):
  File "C:\Users\Sha mofei\PycharmProjects\pythonProject\数据读取\gpt坐标系转换.py", line 27, in <module>
    process_df = tp.trajectory_data(_type='df')
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\gotrackit\gps\LocGps.py", line 732, in trajectory_data
    export_trajectory = pd.merge(export_trajectory, self.__user_gps_info,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 184, in merge
    return op.get_result(copy=copy)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 886, in get_result
    join_index, left_indexer, right_indexer = self._get_join_info()
                                              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 1151, in _get_join_info
    (left_indexer, right_indexer) = self._get_join_indexers()
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 1125, in _get_join_indexers
    return get_join_indexers(
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 1740, in get_join_indexers
    zipped = zip(*mapped)
             ^^^^^^^^^^^^
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 1737, in <genexpr>
    _factorize_keys(left_keys[n], right_keys[n], sort=sort)
  File "C:\Users\Sha mofei\AppData\Roaming\Python\Python312\site-packages\pandas\core\reshape\merge.py", line 2561, in _factorize_keys
    llab = rizer.factorize(lk)  # type: ignore[arg-type]
           ^^^^^^^^^^^^^^^^^^^
  File "pandas\\_libs\\hashtable_class_helper.pxi", line 3045, in pandas._libs.hashtable.Int64Factorizer.factorize
ValueError: Buffer dtype mismatch, expected 'const int64_t' but got 'int'

Process finished with exit code 1
如何解决

**Comment by zdsjjtTLG** [comment]
这个是小概率BUG，等0.3.15发版，就今天

**Reply by zdsjjtTLG** [reply]
或者你先直接改源码：LocGps.py的732行前面加上：

export_trajectory['seq'] = export_trajectory['seq'].astype(np.int64)
self.__user_gps_info['seq'] = self.__user_gps_info['seq'].astype(np.int64)

**Reply by Murphy666-code** [reply]
不好意思在哪加 没看明白 下面是我现在运行的代码：
import os
import pandas as pd
from gotrackit.gps.Trajectory import TrajectoryPoints

if __name__ == '__main__':
    gps_df = pd.read_csv(r'D:\1月6日数据_修改后转坐标-数据清洗.csv')

    # 去除同一出行中的相同定位时间点数据
    gps_df.drop_duplicates(subset=['agent_id', 'time'], keep='first', inplace=True)
    gps_df.reset_index(inplace=True, drop=True)

    # 构建TrajectoryPoints类, 并且指定一个plain_crs
    tp = TrajectoryPoints(gps_points_df=gps_df, time_unit='ms', plain_crs='EPSG:32649')

    # 间隔3个点采样一个点
    # tp.lower_frequency(lower_n=3)

    # 卡尔曼滤波平滑
    tp.kf_smooth()

    # 使用链式操作自定义预处理的先后顺序, 只要保证kf_smooth()操作后没有执行 - 滑动窗口平滑、增密，处理后的轨迹数据即可得到分项速度数据
    # tp.simplify_trajectory().del_dwell_points()
    # tp.kf_smooth().del_dwell_points()

    # 获取清洗后的结果
    # _type参数可以取值为 df 或者 gdf
    process_df = tp.trajectory_data(_type='df')

    out_fldr = r'./data/output/'

    # 存储结果
    process_df.to_csv(os.path.join(out_fldr, r'D:\1月6日数据_修改后转坐标-数据清洗1.csv'), encoding='utf_8_sig', index=False)

    # 输出为html进行动态可视化
    tp.export_html(out_fldr=out_fldr, file_name='sample', radius=9.0)

**Reply by zdsjjtTLG** [reply]
那你等我今天发布新版本把

**Reply by Murphy666-code** [reply]
ok

**Reply by zdsjjtTLG** [reply]
去这个文件的报错的代码行前面加
![1735186151464](https://github.com/user-attachments/assets/7e9378d1-a790-4bc1-8476-9cc8996d4f49)

**Reply by Murphy666-code** [reply]
成功啦 但是结果出了好多行这个：User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')
User Guide: https://docs.kepler.gl/docs/keplergl-jupyter
FileNotFoundError(2, 'No such file or directory')

**Reply by zdsjjtTLG** [reply]
因为你指定的输出目录out_fldr不存在，你得新建一个文件夹
